services/spar/src/Spar/Error.hs

```diff
@@ -188,3 +179,31 @@ renderSparError (SAML.CustomError SparProvisioningTokenLimitReached) = Right $ W
 renderSparError (SAML.CustomError (SparScimError err)) = Left $ Scim.scimToServerError err
 -- Other
 renderSparError (SAML.CustomServant err) = Left err
+
+-- | If a call to another backend service fails, just respond with whatever it said.
+--
+-- FUTUREWORK: with servant, there will be a way for the type checker to confirm that we
+-- handle all exceptions that brig can legally throw!
+rethrow :: LText -> ResponseLBS -> (HasCallStack, Log.MonadLogger m, MonadError SparError m) => m a
+rethrow serviceName resp = do
+  Log.err
+    ( Log.msg ("rfc error" :: Text)
+        . Log.field "status" (Bilge.statusCode resp)
+        . Log.field "error" (show err)
+        . Log.field "callstack" (prettyCallStack callStack)
+    )
+  throwError err
+  where
+    err :: SparError
+    err =
+      responseJsonMaybe resp
+        & maybe
+          ( SAML.CustomError
+              . SparCouldNotParseRfcResponse serviceName
+              . ("internal error: " <>)
+              . cs
```

Comment:
```suggestion
              . cs
              . show
              . (statusCode resp,)
```

(Or did you drop that on purpose?)

Reply:
No, I probably deleted that on accident. Re-added